lib/galaxy/schema/invocation.py

```diff
@@ -214,3 +231,114 @@ class InvocationMessageResponseModel(BaseModel):
 
     class Config:
         orm_mode = True
+
+
+class InvocationState(str, Enum):
+    NEW = "new"  # Brand new workflow invocation... maybe this should be same as READY
+    READY = "ready"  # Workflow ready for another iteration of scheduling.
+    SCHEDULED = "scheduled"  # Workflow has been scheduled.
+    CANCELLED = "cancelled"
+    FAILED = "failed"
+
+
+class InvocationStepState(str, Enum):
+    NEW = "new"  # Brand new workflow invocation step
+    READY = "ready"  # Workflow invocation step ready for another iteration of scheduling.
+    SCHEDULED = "scheduled"  # Workflow invocation step has been scheduled.
+    # CANCELLED = 'cancelled',  TODO: implement and expose
+    # FAILED = 'failed',  TODO: implement and expose
+
+
+class InvocationStepOutput(Model):
+    src: str = Field(
+        literal_to_value(INVOCATION_STEP_OUTPUT_SRC),
+        title="src",
+        description="The source model of the output.",
+        const=True,
+        mark_required_in_schema=True,
+    )
+    id: DecodedDatabaseIdField = Field(
+        ...,
+        title="Dataset ID",
+        description="Dataset ID of the workflow step output.",
+    )
+    uuid: Optional[schema.UUID4] = Field(
+        None,
+        title="UUID",
+        description="Universal unique identifier of the workflow step output dataset.",
+    )
+
+
+class InvocationStepCollectionOutput(Model):
+    src: str = Field(
+        literal_to_value(INVOCATION_STEP_COLLECTION_OUTPUT_SRC),
+        title="src",
+        description="The source model of the output.",
+        const=True,
+        mark_required_in_schema=True,
+    )
+    id: DecodedDatabaseIdField = Field(
```

Comment:
I know at some point I tried to make all encoded IDs be of type `DecodedDatabaseIdField`, but we decided that it was more explicit to use `DecodedDatabaseIdField` for parameters and body payloads (you will get the decoded value automatically in your route handler), and `EncodedDatabaseIdField` for responses (it will ensure you have an encoded value) so it was more explicit.

So in summary, is recommended to use `DecodedDatabaseIdField` for any parameter and body payload model and `EncodedDatabaseIdField` for any ID in a response model.